clang/lib/AST/ExprConstant.cpp

```diff
@@ -2803,9 +2803,10 @@ static bool handleIntIntBinOp(EvalInfo &Info, const Expr *E, const APSInt &LHS,
     // C++11 [expr.shift]p1: Shift width must be less than the bit width of
     // the shifted type.
     unsigned SA = (unsigned) RHS.getLimitedValue(LHS.getBitWidth()-1);
-    if (SA != RHS) {
+    if (SA != RHS && Info.getLangOpts().CPlusPlus11) {
```

Comment:
Why are we removing this diagnostic in C++98 and C?

Reply:
As I didn't find valid reason, I reversed this change.